altair/expr/core.py

```diff
@@ -3,6 +3,8 @@
 import datetime as dt
 from typing import TYPE_CHECKING, Any, Literal, Union
 
+import numpy as np
```

Comment:
rather than importing numpy, you could make a little helper `is_numpy_generic` which does
```python
return (np := sys.modules.get('numpy')) is not None and isinstance(val, np.generic)
```
then use
```python
    elif is_numpy_generic(val):
        return repr(val.item())
```
on L55

Reply:
we could (and should!) add/change this in Narwhals, but given that it's a one-liner, I think you'd be better off just defining a helper function here than bumping the minimum narwhals version

Reply:
Thank you for the tip @MarcoGorelli, I updated the PR. I could make the signature of `is_numpy_generic` more precise by making the return type a `TypeGuard`, but that would mean that we have to import numpy in the `if TYPE_CHECKING` block. I don't think it's worth it, but me know if you'd prefer to add it @mattijn.